test/account/ReplaceModule.t.sol

```diff
@@ -0,0 +1,170 @@
+        account1.executeWithAuthorization(
+            abi.encodeCall(IStandardExecutor.execute, (address(this), 1, callData)),
+            _encodeSignature(
+                ModuleEntityLib.pack(address(singleSignerValidation), TEST_DEFAULT_VALIDATION_ENTITY_ID),
+                GLOBAL_VALIDATION,
+                ""
+            )
+        );
+
+        // upgrade module by batching uninstall + install calls
+        bytes[] memory emptyBytesArr = new bytes[](0);
+        Call[] memory calls = new Call[](2);
+        calls[0] = Call({
+            target: address(account1),
+            value: 0,
+            data: abi.encodeCall(
+                IModuleManager.uninstallValidation,
+                (currModuleEntity, abi.encode(TEST_DEFAULT_VALIDATION_ENTITY_ID), emptyBytesArr)
+            )
+        });
+        calls[1] = Call({
+            target: address(account1),
+            value: 0,
+            data: abi.encodeCall(
+                IModuleManager.installValidation,
+                (
+                    ValidationConfigLib.pack(newModuleEntity, true, true),
+                    new bytes4[](0),
+                    abi.encode(TEST_DEFAULT_VALIDATION_ENTITY_ID + 1, owner1),
+                    new bytes[](0)
+                )
+            )
+        });
+        account1.executeWithAuthorization(
+            abi.encodeCall(account1.executeBatch, (calls)),
+            _encodeSignature(
+                ModuleEntityLib.pack(address(singleSignerValidation), TEST_DEFAULT_VALIDATION_ENTITY_ID),
+                GLOBAL_VALIDATION,
+                ""
+            )
+        );
+
+        // Test if old validation still works, expect fail
+        vm.expectRevert(
+            abi.encodePacked(
+                UpgradeableModularAccount.ValidationFunctionMissing.selector,
+                abi.encode(IStandardExecutor.execute.selector)
+            )
+        );
+        account1.executeWithAuthorization(
+            abi.encodeCall(IStandardExecutor.execute, (address(this), 1, callData)),
+            _encodeSignature(
+                ModuleEntityLib.pack(address(singleSignerValidation), TEST_DEFAULT_VALIDATION_ENTITY_ID),
+                GLOBAL_VALIDATION,
+                ""
+            )
+        );
+
+        // Test if new validation works
+        vm.expectEmit(true, true, true, true);
+        emit ReceivedTestCall(callData, 1);
+        account1.executeWithAuthorization(
+            abi.encodeCall(IStandardExecutor.execute, (address(this), 1, callData)),
+            _encodeSignature(
+                ModuleEntityLib.pack(address(validation2), TEST_DEFAULT_VALIDATION_ENTITY_ID + 1),
+                GLOBAL_VALIDATION,
+                ""
+            )
+        );
+    }
+
+    event ReceivedTestCall(bytes, uint256);
+
+    fallback() external payable {
+        emit ReceivedTestCall(msg.data, msg.value);
+    }
+
+    receive() external payable {
+        emit ReceivedTestCall("", msg.value);
+    }
```

Comment:
Do we need this?

Reply:
Using it to check execution success. Actually I can simplify by checking address.balance instead